modules/form/src/addons/scopes.tsx

```diff
@@ -0,0 +1,249 @@
+                     value: buildScopesString(scopes)
+                 }
+             } as React.ChangeEvent<HTMLInputElement>
+        );
+    };
+
+    /**
+      * Update input field values for scope.
+      * 
+      * @param scope - Scope.
+      */
+    const updateScopeInputFields = (scope: ScopeInterface) => {
+        setScopeValue(scope?.value);
+    };
+
+    /**
+     * Enter button option.
+     * @param e - keypress event.
+     */
+    const keyPressed = (e: React.KeyboardEvent<HTMLInputElement>): void => {
+ 
+        if (e.key === "Enter" ) {
+            handleScopeAdd(e);
+        }
+    };
+
+    const handleScopeAdd = (event) => {
+        event.preventDefault();
+        if (isEmpty(scopeValue)) {
+            return;
+        }
+
+        setErrorMessage("");
+        const output: ScopeInterface[] = [ {
+            value: scopeValue
+        } ];
+ 
+        scopes.forEach(function(scope) {
+            const existing = output.filter((item) => {
+                return item.value == scope.value;
+            });
+ 
+            if (existing.length) {
+                return;
+            } else {
+                output.push(scope);
+            }
+        });
+ 
+        setScopes(output);
+        updateScopeInputFields({
+            value: ""
+        });
+    };
+ 
+    const handleLabelRemove = (scopeParam: string) => {
+
+        if (isEmpty(scopeParam)) {
+            return;
+        }
+        setScopes(filter(scopes, scope => !isEqual(scope,
+            buildScope(scopeParam))));
+    };
+
+    return (
+        <>
+            <Form.Group grouped={ true }>
+                {
+                    label && (
+                        <div className={ `field ${ required ? "required" : "" }` }>
+                            <label>{ label }</label>
+                        </div>
+                    )
+                }
+                <Form.Group inline widths="equal" unstackable={ true }> 
+                    <Form.Input
+                        fluid
+                        value={ scopeValue }
+                        onBlur={ onBlur }
+                        focus
+                        onChange={ (event, data) => {
+                            setScopeValue(data.value.trim());
+                        } }
+                        onKeyDown={ keyPressed }
+                        placeholder = { placeholder }
+
+                    />
+                    <Popup
+                        trigger={
+                            (
+                                <Button
+                                    onClick={ (e) => handleScopeAdd(e) }
+                                    icon="add"
+                                    type="button"
+                                    disabled={ false }
+                                />
+                            )
+                        }
+                        position="top center"
+                        content="Add scope"
```

Comment:
Resolve these texts from a translations object.

```jsx
content={ translations.addScope ?? "Add scope" }
```

Ref: https://github.com/wso2/identity-apps/blob/9c3dcb8a6e0376c39dc9606c24d21c113646f52f/modules/react-components/src/components/modal/edit-avatar-modal/edit-avatar-modal.tsx#L99

Reply:
Added a TODO as this has to be injected through multiple modules